tests/unit_tests/test_filters.py

```diff
@@ -0,0 +1,148 @@
+    assert elem.find('order').text == str(n)
+
+
+def test_spatial_legendre():
+    n = 5
+    axis = 'x'
+    f = openmc.SpatialLegendreFilter(n, axis, -10., 10.)
+    assert f.order == n
+    assert f.axis == axis
+    assert f.minimum == -10.
+    assert f.maximum == 10.
+    assert f.bins[0] == 'P0'
+    assert f.bins[-1] == 'P5'
+    assert len(f.bins) == n + 1
+
+    # Make sure __repr__ works
+    repr(f)
+
+    # to_xml_element()
+    elem = f.to_xml_element()
+    assert elem.tag == 'filter'
+    assert elem.attrib['type'] == 'spatiallegendre'
+    assert elem.find('order').text == str(n)
+    assert elem.find('axis').text == str(axis)
+
+
+def test_spherical_harmonics():
+    n = 3
+    f = openmc.SphericalHarmonicsFilter(n)
+    f.cosine = 'particle'
+    assert f.order == n
+    assert f.bins[0] == 'Y0,0'
+    assert f.bins[-1] == 'Y{0},{0}'.format(n, n)
+    assert len(f.bins) == (n + 1)**2
+
+    # Make sure __repr__ works
+    repr(f)
+
+    # to_xml_element()
+    elem = f.to_xml_element()
+    assert elem.tag == 'filter'
+    assert elem.attrib['type'] == 'sphericalharmonics'
+    assert elem.attrib['cosine'] == f.cosine
+    assert elem.find('order').text == str(n)
+
+
+def test_zernike():
+    n = 4
+    f = openmc.ZernikeFilter(n, 0., 0., 1.)
+    assert f.order == n
+    assert f.bins[0] == 'Z0,0'
+    assert f.bins[-1] == 'Z{0},{0}'.format(n)
+    assert len(f.bins) == (n + 1)*(n + 2)//2
+
+    # Make sure __repr__ works
+    repr(f)
+
+    # to_xml_element()
+    elem = f.to_xml_element()
+    assert elem.tag == 'filter'
+    assert elem.attrib['type'] == 'zernike'
+    assert elem.find('order').text == str(n)
+
+
+def test_first_moment(run_in_tmpdir, box_model):
+    plain_tally = openmc.Tally()
+    plain_tally.scores = ['flux', 'scatter']
+
+    # Create tallies with expansion filters
+    leg_tally = openmc.Tally()
+    leg_tally.filters = [openmc.LegendreFilter(3)]
+    leg_tally.scores = ['scatter']
+    leg_sptl_tally = openmc.Tally()
+    leg_sptl_tally.filters = [openmc.SpatialLegendreFilter(3, 'x', -5., 5.)]
+    leg_sptl_tally.scores = ['scatter']
+    sph_scat_filter = openmc.SphericalHarmonicsFilter(5)
+    sph_scat_filter.cosine = 'scatter'
+    sph_scat_tally = openmc.Tally()
+    sph_scat_tally.filters = [sph_scat_filter]
+    sph_scat_tally.scores = ['scatter']
+    sph_flux_filter = openmc.SphericalHarmonicsFilter(5)
+    sph_flux_filter.cosine = 'particle'
+    sph_flux_tally = openmc.Tally()
+    sph_flux_tally.filters = [sph_flux_filter]
+    sph_flux_tally.scores = ['flux']
+    zernike_tally = openmc.Tally()
+    zernike_tally.filters = [openmc.ZernikeFilter(3, r=10.)]
+    zernike_tally.scores = ['scatter']
+
+    # Add tallies to model and ensure they all use the same estimator
+    box_model.tallies = [plain_tally, leg_tally, leg_sptl_tally,
+                         sph_scat_tally, sph_flux_tally, zernike_tally]
+    for t in box_model.tallies:
+        t.estimator = 'analog'
```

Comment:
Yes, the estimator is changed automatically if nothing is specified. The purpose of forcing everything to use an analog estimator is so that the tallies with expansion filters have the same answers as the "plain" tally.